ci/azure-test-all.yml

```diff
@@ -0,0 +1,25 @@
+steps:
+- checkout: self
+  fetchDepth: 1
+
+- template: azure-install-rust.yml
+
+- bash: rustup target add $OTHER_TARGET
+  displayName: "Install cross-compile target"
+
+- bash: sudo apt install gcc-multilib
+  displayName: "Install gcc-multilib (linux)"
+  condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'))
+
+# Some tests rely on a clippy command to run, so let's try to install clippy to
+# we can be sure to run those tests.
+- bash: rustup component add clippy || echo "clippy not available"
+  displayName: "Install clippy (maybe)"
+
+# Deny warnings on CI to avoid warnings getting into the codebase, and note the
+# `force-system-lib-on-osx` which is intended to fix compile issues on OSX where
+# compiling curl from source on OSX yields linker errors on Azure. It's sort of
+# unclear how to fix these linker errors and why it worked previously on Travis,
+# but if you can remove the `curl` feature please do.
+- bash: cargo test --features 'deny-warnings curl/force-system-lib-on-osx'
```

Comment:
I did some investigation and my results are at https://github.com/alexcrichton/curl-rust/issues/279.

The reason it works on Travis is because it is using the real system curl, with the HTTP2 feature built-in.  It looks like azure installs their own curl (`/usr/local/opt/curl`) which does not have HTTP2 enabled which fails [this test](https://github.com/alexcrichton/curl-rust/blob/0ba03a4790710425515e8177591816d185ad73a6/curl-sys/build.rs#L447), which causes it to be built statically, which hits the bug listed above.

Reply:
Thanks for this! I've updated the comment here to reflect the results of what you found.

FWIW I think that's why things worked by default on Travis but IIRC the curl-rust crate still tested compiling libcurl statically and it ended up working out, but it may have used different surface areas than Cargo itself.